File: 프로그래머스/lv1/86491. 최소직사각형/최소직사각형.py
# def solution(sizes):
#     #사이즈 길이 1~10000 /w,h 형식 1~1000
#     answer = 1
    
#     for i in sizes:
#         i.sort( reverse=True)
#     # print(sizes)
        
#     sizes.sort(key=lambda x:x[0], reverse=True)
#     # print(sizes)
#     answer= answer* sizes[0][0]
    
#     sizes.sort(key=lambda x:x[1], reverse=True)
#     answer= answer* sizes[0][1]
    
#     return answer

def solution(sizes):
    
    answer=0
    for i in sizes:
        i.sort()
    print(sizes)
    short=0
    long=0
    for j in sizes:
        if j[0]>=short:
            short=j[0]
        if j[1]>=long:
            long=j[1]
    answer=short*long
    return answer

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.js
function solution(sizes) {
    let answer = 0;
    let x=0
    let y=0
    for( s of sizes ){
        s.sort((a,b)=>a-b)
        if (s[0]>x){x=s[0]}
        if (s[1]>y){y=s[1]}
    }
    answer=x*y
    return answer;
}